### PR TITLE
add desktop support, refactor scoring data

### PR DESCRIPTION
**Pull request by connorjclark**
I started work on getting different scoring curve based on devices to work, but first had to refactor much of the scoring data.

I got to the point where I realized we need to support the new scoring curve definition. Need to pick this up another time.

### CODE REVIEW

File: script/metrics.js [hunk 1]
```diff
@@ -1,36 +1,60 @@
...
+    FMP: {weight: 0.066666, median: 4000, falloff: 2000, auditId: 'first-meaningful-paint', name: 'First Meaningful Paint'},
+    TTI: {weight: 0.33333, median: 7300, falloff: 2900, auditId: 'interactive', name: 'Time to Interactive'},
+    FCI: {weight: 0.133333, median: 6500, falloff: 2900, auditId: 'first-cpu-idle', name: 'First CPU Idle'},
   },
+};
+
+// TODO need to support defining curves with p10 and podr.
+
+const scoringGuides_WIP_mobile = {
   v6: {
```
Comment (paulirish) [inline script/metrics.js:52]: I don't love how WET this file is now. :/  

we could compose objects like this pretty easily i think.

how about this file starts with something like

```js
const weights = {
  v5: {
    FCP: 0.2,
    SI: 0.26666,
    FMP: 0.066666,
    TTI: 0.33333,
    FCI: 0.133333,
  },
  v6: {
    FCP: 0.15,
    SI:  0.15,
    LCP: 0.25,
    TTI: 0.15,
    TBT: 0.25,
    CLS: 0.05
  },
};

const audits = {
  FCP: {auditId: 'first-contentful-paint', name: 'First Contentful Paint'},
  FMP: {auditId: 'first-meaningful-paint', name: 'First Meaningful Paint'},
  SI:  {auditId: 'speed-index', name: 'Speed Index'},
  TTI: {auditId: 'interactive', name: 'Time to Interactive'},
  FCI: {auditId: 'first-cpu-idle', name: 'First CPU Idle'},
  TBT: {auditId: 'total-blocking-time', name: 'Total Blocking Time'},
  LCP: {auditId: 'largest-contentful-paint', name: 'Largest Contentful Paint'},
  CLS: {auditId: 'cumulative-layout-shift', name: 'Cumulative Layout Shift', units: 'unitless'},
};


const curves = {
    FCP: {mobile: { median: 4000, falloff: 2000,  }, desktop: { median: 4000, falloff: 2000, } },
    FMP: {mobile: { median: 4000, falloff: 2000,  }, desktop: { median: 4000, falloff: 2000, } },
    SI: {mobile: { median: 5800, falloff: 2900,  },  desktop: { median: 5800, falloff: 2900, } },
    TTI: {mobile: { median: 7300, falloff: 2900,  }, desktop: { median: 7300, falloff: 2900, } },
    FCI: {mobile: { median: 6500, falloff: 2900,  }, desktop: { median: 6500, falloff: 2900, } },
    TBT: {mobile: { median: 600, falloff: 200,  },   desktop: { median: 600, falloff: 200, } },
    LCP: {mobile: { median: 4000, falloff: 2000,  }, desktop: { median: 4000, falloff: 2000, } },
    CLS: {mobile: { median: 0.25, falloff: 0.054, }, desktop: { median: 0.25, falloff: 0.054,} },
};
```

and then composes some scoringGuide object at the end.

File: script/main.js [hunk 2]
```diff
@@ -220,7 +222,10 @@ class App extends Component {
     // debounce just a tad, as its noisy
     debounce(_ => {
       const url = new URL(location.href);
-      const auditIdValuePairs = Object.entries(this.state).map(([id, value]) => [scoring[id].auditId,value]);
+      const auditIdValuePairs = Object.entries(this.state).map(([id, value]) => {
+        const auditId = (scoringGuides.v5[id] || scoringGuides.v6[id]).auditId;
```
Comment (paulirish) [inline script/main.js]: this is a bit awkward.

**Review by paulirish — COMMENTED** [review]
i left some ideas on what metrics.js looks like...



> I got to the point where I realized we need to support the new scoring curve definition. 

as for p10 scoring, there's two approaches we can take 1)  support both p10-style & podr style and have the data determine. or 2) we only support p10 style and fix the data to match. 

i'm good with whatever

File: script/main.js (hunk 2 above)
Comment (connorjclark) [reply script/main.js]: yeaeh, let's make a metrics object that is just the metric name, audit id.... exactly what you named `audits`. 👍

**Comment by connorjclark** [comment]
> 1) support both p10-style & podr style and have the data determine. 

I'd prefer this

**Comment by connorjclark** [comment]
scoring curves now derive podr from p10.

verify it's all cool with:

```
http://localhost:8000/#first-contentful-paint=4143.975&speed-index=8527.592402791312&largest-contentful-paint=8930.763000000003&interactive=20955.31199999999&total-blocking-time=2375.5&cumulative-layout-shift=0.134375&first-cpu-idle=11466.559000000001&first-meaningful-paint=6435.7245&device=mobile&version=6.0.0&version=5.6.0
```

image

image

**Comment by paulirish** [comment]
ruhroh. 

https://github.com/GoogleChrome/lighthouse/issues/10836 has a WPT report where "device" is 'none' as it's not using desktop emulation. (But it is a desktop run)

We'll have to figure out how we handle that case.

**Comment by connorjclark** [comment]
Shouldn't we just use the desktop curves for emulation none? Is that what LH does?

**Comment by paulirish** [comment]
Yeah for  sure

**Comment by patrickhulce** [comment]
> Shouldn't we just use the desktop curves for emulation none? Is that what LH does?

Problem: when you run WPT on an actual phone it also uses `none` and we definitely don't want to use the desktop curves there :/ We might need to update the report to infer from the UA or use `TestedAsMobileDevice` or something when device emulation is none.

Example test https://webpagetest.org/result/200524_3N_9f238902d8910c6a5dd67ad3a05ab1eb/

**Comment by patrickhulce** [comment]
Honestly maybe the better message is to say "warning: this test didn't match the two lighthouse presets" and just have a toggle for mobile v. desktop?

**Comment by fvpDev** [comment]
> Honestly maybe the better message is to say "warning: this test didn't match the two lighthouse presets" and just have a toggle for mobile v. desktop?

As a user, I would prefer this^...In general, having a toggle for that would be nice. Though in the context of wpt, they are already separating mobile tests from desktop tests, thus already separating LH runs. I think on the wpt channel it would be could to keep them separated as they do; and not have a toggle there, so as to avoid potential confusion. However on web.dev, the toggle would be nice to have for sure...currently right now it is an Emulated Nexus 5X when running in desktop view, which is a big no no.

**Comment by connorjclark** [comment]
Let's just add a toggle to the scoring calculator. if it's none, just default to _something_ (it'd be wrong in some cases for whatever we pick, so let's just pick mobile?)

we determine the scoring guide in LH w/ `artifacts.TestedAsMobileDevice`, which is a mix of emulated form factor/hosted device, so perhaps we just need to surface that in the LHR and use that in the report calc URL.

in any case, let's land this PR and follow up with a device toggle.

**Comment by benschwarz** [comment]
> Let's just add a toggle to the scoring calculator. if it's none, just default to something (it'd be wrong in some cases for whatever we pick, so let's just pick mobile?)

@connorjclark Isn't `none` always going to be a Desktop run? (Unless the user has purposefully set their own screen dimensions/throttling/etc etc)

**Comment by connorjclark** [comment]
> @connorjclark Isn't none always going to be a Desktop run? (Unless the user has purposefully set their own screen dimensions/throttling/etc etc)

exactly :)

see wpt: this code runs for both mobile or desktop host device. it doesn't configure what type it is–we derive that in LH via `TestedOnMobileDevice`
https://github.com/connorjclark/wptagent/blob/master/internal/devtools_browser.py#L490

**Review by paulirish — APPROVED** [review]
Double-checked some v6 and v5 scores and they're all looking great.

Agree with following up this PR with a device toggle.

**Comment by paulirish** [comment]
This is deployed

**Pull request merged**